lib/openxml/has_properties.rb

```diff
@@ -142,6 +152,18 @@ def property_xml(xml)
       end
     end
 
+    def build_required_properties
+      required_properties.each do |prop|
+        public_send(:"#{prop}=", default_property_value_for(prop)) if respond_to? :"#{prop}="
+        property_instance = public_send(:"#{prop}")
+        property_instance.build_required_properties if property_instance.respond_to?(:build_required_properties)
+      end
+    end
```

Comment:
What calls `build_required_properties`?

Do you (the programmer using this) have to know to call `build_required_properties` sometime later whenever you define `:required` properties?

Reply:
That's what I'd had in mind (and how I'd been using it). The idea is that something _with_ properties (like a presentation or slide part) would either call it as part of their `initialize` or else have it called on them as part of a convenience builder method.

With that in mind, I could see a good argument for this being placed by default in the `initialize` method, perhaps even _before_ calling `super` 🤔

Reply:
I guess part of the argument _against_ putting in `initialize` by default is that it would break some of our automatic testing macros, automatically generating child tags that we aren't looking to test. 😓